gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+    assertNull(created.nameServerSet());
+    assertNull(created.id());
+    created.delete();
+    created = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.DNS_NAME));
+    assertEquals(ZONE1.name(), created.name()); // always returned
+    assertNull(created.creationTimeMillis());
+    assertEquals(ZONE1.dnsName(), created.dnsName());
+    assertNull(created.description());
+    assertTrue(created.nameServers().isEmpty()); // never returns null
+    assertNull(created.nameServerSet());
+    assertNull(created.id());
+    created.delete();
+    created = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.NAME));
+    assertEquals(ZONE1.name(), created.name()); // always returned
+    assertNull(created.creationTimeMillis());
+    assertNull(created.dnsName());
+    assertNull(created.description());
+    assertTrue(created.nameServers().isEmpty()); // never returns null
+    assertNull(created.nameServerSet());
+    assertNull(created.id());
+    created.delete();
+    created = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.NAME_SERVER_SET));
+    assertEquals(ZONE1.name(), created.name()); // always returned
+    assertNull(created.creationTimeMillis());
+    assertNull(created.dnsName());
+    assertNull(created.description());
+    assertTrue(created.nameServers().isEmpty()); // never returns null
+    assertNull(created.nameServerSet()); // we did not set it
+    assertNull(created.id());
+    created.delete();
+    created = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.NAME_SERVERS));
+    assertEquals(ZONE1.name(), created.name()); // always returned
+    assertNull(created.creationTimeMillis());
+    assertNull(created.dnsName());
+    assertNull(created.description());
+    assertFalse(created.nameServers().isEmpty());
+    assertNull(created.nameServerSet());
+    assertNull(created.id());
+    created.delete();
+    created = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.ZONE_ID));
+    assertEquals(ZONE1.name(), created.name()); // always returned
+    assertNull(created.creationTimeMillis());
+    assertNull(created.dnsName());
+    assertNull(created.description());
+    assertNotNull(created.nameServers());
+    assertTrue(created.nameServers().isEmpty()); // never returns null
+    assertNotNull(created.id());
+    created.delete();
+    // combination of multiple things
+    created = DNS.create(ZONE1, Dns.ZoneOption.fields(Dns.ZoneField.ZONE_ID,
+        Dns.ZoneField.NAME_SERVERS, Dns.ZoneField.NAME_SERVER_SET, Dns.ZoneField.DESCRIPTION));
+    assertEquals(ZONE1.name(), created.name()); // always returned
+    assertNull(created.creationTimeMillis());
+    assertNull(created.dnsName());
+    assertEquals(ZONE1.description(), created.description());
+    assertFalse(created.nameServers().isEmpty());
+    assertNull(created.nameServerSet()); // we did not set it
+    assertNotNull(created.id());
+  }
+
+  @Test
+  public void testZoneReload() {
```

Comment:
I am not sure we need integration tests for this, we already have unit tests for Zone that mock the dns service and integration tests for `dns.getZone`